### DISCUSSION TITLE
How to build mumps in Windows 10 + VS2019 + intel Fortran 2020?

### DISCUSSION THREAD

**Discussion by TangLaoya** [scivision/mumps-superbuild#47] [Q&A] [answered]
Dear all,
Thanks for providing the useful cmake files to help build mumps in Windows. I noticed that the cmake files can download necessary source code automatically. But after I download the whole project here and try to build mumps by cmake, it displayed the following error messages:

`Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
The C compiler identification is MSVC 19.29.30146.0
The Fortran compiler identification is Intel 19.1.3.20201010
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
Detecting C compile features
Detecting C compile features - done
Detecting Fortran compiler ABI info
Detecting Fortran compiler ABI info - done
Determine Intel Fortran Compiler Implicit Link Path
Determine Intel Fortran Compiler Implicit Link Path - done
Check for working Fortran compiler: C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2020/windows/bin/intel64/ifort.exe - skipped
MUMPS 5.5.1.8  CMake 3.24.0-rc2  Toolchain 
checking that C and Fortran compilers can link
checking that C and Fortran compilers can link - OK
Performing Test CMAKE_HAVE_LIBC_PTHREAD
Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
Looking for pthread_create in pthreads
Looking for pthread_create in pthreads - not found
Looking for pthread_create in pthread
Looking for pthread_create in pthread - not found
Check if compiler accepts -pthread
Check if compiler accepts -pthread - no
Found Threads: TRUE  
CMake Error at C:/Program Files/CMake/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find LAPACK (missing: LAPACK_LIBRARY LAPACK_links)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindLAPACK.cmake:483 (find_package_handle_standard_args)
  cmake/lapack.cmake:22 (find_package)
  CMakeLists.txt:39 (include)

Configuring incomplete, errors occurred!
See also "J:/GitHub/mumps/64bit/CMakeFiles/CMakeOutput.log".
See also "J:/GitHub/mumps/64bit/CMakeFiles/CMakeError.log".`

Is there anything I missed?

I just simply fill the source code path and build path in cmake IDE, and then click 'Configure' button.

Thanks,
Tang Laoya

**Comment by scivision** [comment]
On Windows, please use oneAPI with Base Toolkit and HPC toolkit. You must use oneAPI's built-in LAPACK and Scalpaack, building Lapack and Scalapack with oneAPI will fail.

**Comment by TangLaoya** [comment]
Dear scivision,
Thank you very much for your kindly reply. That's great, it can download source code when I open the project by VS and build the project. Could you please also integrate the /scripts/CMakesLists.txt into the main CMakeLists.txt, and specify the built libs into given path?

Thanks,
Tang Laoya

**Reply by scivision** [reply]
I don't integrate the external libraries build into the main CMakeLists.txt for robustness and simplicity. For example, with oneAPI trying to build Lapack and Scalapack fails.

**Comment by TangLaoya** [comment]
In addition, could you please also set an option to check whether users have installed intel MKL? If users have installed intel MKL, they could use intel MKL instead of the LAPACK and SCALAPACK.

**Reply by scivision** [reply]
If MKL is installed, the default is to prefer MKL. Ensure that environment variable MKLROOT points to the top directory where oneAPI MKL is installed

**Comment by TangLaoya** [comment]
Dear scivision,
I have successfully downloaded lapack, metis, scalapack and scotch in VS when build the project, but only the metis are successfully built and the metis.lib are put in MumpsPrereqs\lib. 

What have I missed?

Thanks,
Tang Laoya

**Reply by scivision** [reply]
I would try to make the plain MUMPS work before trying fancy options like Scotch. At least, you need to specify the location of Scotch to MUMPS cmake, from the MUMPS folder like:

```sh
cmake -Bbuild -DScotch_ROOT=/path/to/scotch
```

**Comment by scivision** [comment]
For Windows, we recommend using Intel oneAPI instead of mixing with Visual Studio as it's easier to use oneAPI alone on Windows. There is a new [Windows Readme](https://github.com/scivision/mumps/blob/main/Readme_Windows.md)

**Comment by szl-opensees** [comment]
try open the powershell and run "cmd.exe "/K" '"C:\Program Files (x86)\Intel\oneAPI\setvars.bat" && powershell'"

**Comment by 1lihua** [comment]
Did you install mumps in windows successfully? I'm now learning mumps. I would like to ask you some questions.

**Reply by scivision** [reply, marked as answer]
Yes: https://github.com/scivision/mumps/blob/main/Readme_Windows.md

**Reply by 1lihua** [reply]
Thank you for your kind reply. I have built mumps in windows successfully.
